faq-bundle/src/EventListener/SitemapListener.php

```diff
@@ -57,35 +56,27 @@ public function __invoke(SitemapEvent $event): void
                 continue;
             }
 
-            // Get the URL of the jumpTo page
-            if (!isset($arrProcessed[$objFaq->jumpTo])) {
-                $objParent = $this->framework->getAdapter(PageModel::class)->findWithDetails($objFaq->jumpTo);
+            $objParent = $this->framework->getAdapter(PageModel::class)->findWithDetails($objFaq->jumpTo);
```

Comment:
Afair `PageModel::findWithDetaild` removed it from the registry so this would be quite the performance penalty. Maybe we should still cache the jumpTo object instead?

Reply:
Unable to reproduce. The numer of DB queries before and after this change didn’t change in my test.

> Afair `PageModel::findWithDetaild` removed it from the registry

What do you mean by that?

Reply:
PageModel::loadDetails()` calls `preventSaving` which removes it from the registry. https://github.com/contao/contao/blob/5.x/core-bundle/contao/models/PageModel.php#L1067
not sure if I‘m missing something?

Reply:
> PageModel::loadDetails() calls preventSaving which removes it from the registry.

No, it doesn’t.

Reply:
@aschempp `$model->preventSaving()` and `$model->preventSaving(true)` keeps a clone in the registry. Only if you call `$model->preventSaving(false)` will the model be removed from the registry.

Reply:
true, I didn't notice that. But it does not keep the details data! So for every page model fetched, it will re-iterate the page tree (without DB queries, because all parents are cached as well) and call the `loadDetails` method. Feels pretty wrong to me?

Reply:
See https://github.com/contao/contao/issues/5153

Reply:
For the record: `findWithDetails()` only gets called once per FAQ category, not once per FAQ item.

Reply:
Added a page cache for `ModuleFaqList` and `ModuleNewsletterList` in 96008592a54300a60a7e68dde72c51cc22ae4f38